### PR TITLE
openstack: add size field to image inventory

### PR DESCRIPTION
**Pull request by bennyz**
- [x] Use size when virtual_size is missing

### CODE REVIEW

**Review by ahadas — APPROVED** [review]

File: pkg/controller/plan/adapter/openstack/builder.go [hunk 1]
```diff
@@ -229,9 +229,14 @@ func (r *Builder) PodEnvironment(_ ref.Ref, _ *core.Secret) (env []core.EnvVar,
 func (r *Builder) PersistentVolumeClaimWithSourceRef(da interface{}, storageName *string, populatorName string, accessModes []core.PersistentVolumeAccessMode, volumeMode *core.PersistentVolumeMode) *core.PersistentVolumeClaim {
 	image := da.(*openstack.Image)
 	apiGroup := "forklift.konveyor.io"
-	size := image.VirtualSize
+	virtualSize := image.VirtualSize
+	// virtual_size may not always be available
+	if virtualSize == 0 {
+		virtualSize = image.Size
```
Comment (mmartinv) [inline pkg/controller/plan/adapter/openstack/builder.go:235]:
```suggestion
		virtualSize = image.SizeBytes
```

File: pkg/controller/provider/container/openstack/resource.go [hunk 2]
```diff
@@ -112,6 +112,7 @@ func (r *Image) ApplyTo(m *model.Image) {
 	m.File = r.File
 	m.Schema = r.Schema
 	m.VirtualSize = r.VirtualSize
+	m.Size = r.SizeBytes
```
Comment (mmartinv) [inline pkg/controller/provider/container/openstack/resource.go:115]: Why not using `m.SizeBytes` directly instead of adding another field?

File: pkg/controller/provider/model/openstack/model.go [hunk 3]
```diff
@@ -78,6 +78,7 @@ type Image struct {
 	File                        string                 `sql:""`
 	Schema                      string                 `sql:""`
 	VirtualSize                 int64                  `sql:""`
+	Size                        int64                  `sql:""`
```
Comment (mmartinv) [inline pkg/controller/provider/model/openstack/model.go:81]: Why is this needed if we already have `SizeBytes`

File: pkg/controller/provider/web/openstack/image.go [hunk 4]
```diff
@@ -45,6 +45,7 @@ type Image struct {
 	File                        string            `json:"file"`
 	Schema                      string            `json:"schema"`
 	VirtualSize                 int64             `json:"virtual_size"`
+	Size                        int64             `json:"size"`
```
Comment (mmartinv) [inline pkg/controller/provider/web/openstack/image.go:48]: Same as above.

File: pkg/controller/provider/web/openstack/image.go [hunk 5]
```diff
@@ -81,6 +82,7 @@ func (r *Image) With(m *model.Image) {
 	r.File = m.File
 	r.Schema = m.Schema
 	r.VirtualSize = m.VirtualSize
+	r.Size = m.Size
```
Comment (mmartinv) [inline pkg/controller/provider/web/openstack/image.go:85]: Same as above

**Review by mmartinv — CHANGES_REQUESTED** [review]

File: pkg/controller/plan/adapter/openstack/builder.go [hunk 6]
```diff
@@ -229,9 +229,14 @@ func (r *Builder) PodEnvironment(_ ref.Ref, _ *core.Secret) (env []core.EnvVar,
 func (r *Builder) PersistentVolumeClaimWithSourceRef(da interface{}, storageName *string, populatorName string, accessModes []core.PersistentVolumeAccessMode, volumeMode *core.PersistentVolumeMode) *core.PersistentVolumeClaim {
 	image := da.(*openstack.Image)
 	apiGroup := "forklift.konveyor.io"
-	size := image.VirtualSize
+	virtualSize := image.VirtualSize
+	// virtual_size may not always be available
```
Comment (ahadas) [inline pkg/controller/plan/adapter/openstack/builder.go:233]: would it make sense to add a validation warning about this?

Comment (bennyz) [reply pkg/controller/plan/adapter/openstack/builder.go:233]: As discussed, we use the size provided by the user anyway

**Review by mmartinv — APPROVED** [review]

File: pkg/controller/provider/web/openstack/image.go [hunk 7]
```diff
@@ -38,7 +38,7 @@ type Image struct {
 	Visibility                  string            `json:"visibility"`
 	Hidden                      bool              `json:"os_hidden"`
 	Checksum                    string            `json:"checksum"`
-	SizeBytes                   int64             `json:"-"`
+	SizeBytes                   int64             `json:"size"`
```
Comment (ahadas) [inline pkg/controller/provider/web/openstack/image.go:41]: how about documenting what it represents - my previous comment assumed it refers to the actual/apparent size..

File: pkg/controller/plan/adapter/openstack/builder.go [hunk 8]
```diff
@@ -229,9 +229,13 @@ func (r *Builder) PodEnvironment(_ ref.Ref, _ *core.Secret) (env []core.EnvVar,
...
+	if virtualSize == 0 {
+		virtualSize = image.SizeBytes
+	}
 	if *volumeMode == core.PersistentVolumeFilesystem {
-		size = int64(float64(image.VirtualSize) * 1.1)
+		virtualSize = int64(float64(image.VirtualSize) * 1.1)
```
Comment (ahadas) [inline pkg/controller/plan/adapter/openstack/builder.go:238]:
```suggestion
		virtualSize = int64(virtualSize) * 1.1)
```
?

**Review by ahadas — APPROVED** [review]
